```
/home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/sycl/TestSYCL_InterOp_Streams.cpp:69:58: error: use of undeclared identifier 'p'
   69 |     cgh.parallel_for(sycl::range<1>(100), [=](int idx) { p[idx] += idx; });
      |                                                          ^
/home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/sycl/TestSYCL_InterOp_Streams.cpp:74:21: error: use of undeclared identifier 'p'
   74 |   queue.memcpy(h_p, p, sizeof(int) * 100);
      |

--- a/core/unit_test/sycl/TestSYCL_InterOp_Init.cpp
+++ b/core/unit_test/sycl/TestSYCL_InterOp_Init.cpp
@@ -17,18 +17,23 @@ namespace Test {
 // SYCL.
 TEST(sycl, raw_sycl_interop) {
   // Make sure all queues use the same context
-  Kokkos::initialize();
-  Kokkos::SYCL default_space;
-  sycl::context default_context = default_space.sycl_queue().get_context();
-
-  sycl::queue queue(default_context, sycl::default_selector_v,
-                    sycl::property::queue::in_order());
+  sycl::queue queue;
+  int* p;
   constexpr int n = 100;
-  int* p          = sycl::malloc_device<int>(n, queue);
+
+  Kokkos::initialize();
   {
-    TEST_EXECSPACE space(queue);
-    Kokkos::View<int*, Kokkos::MemoryTraits<Kokkos::Unmanaged>> v(p, n);
-    Kokkos::deep_copy(space, v, 5);
+    Kokkos::SYCL default_space;
+    sycl::context default_context = default_space.sycl_queue().get_context();
+
+    queue = sycl::queue(default_context, sycl::default_selector_v,
+                        sycl::property::queue::in_order());
+    p     = sycl::malloc_device<int>(n, queue);
+    {
+      TEST_EXECSPACE space(queue);
+      Kokkos::View<int*, Kokkos::MemoryTraits<Kokkos::Unmanaged>> v(p, n);
+      Kokkos::deep_copy(space, v, 5);
+    }
   }
   Kokkos::finalize();
 

--- a/core/unit_test/sycl/TestSYCL_InterOp_Streams.cpp
+++ b/core/unit_test/sycl/TestSYCL_InterOp_Streams.cpp
@@ -8,53 +8,59 @@ namespace Test {
 // Test Interoperability with SYCL Streams
 TEST(sycl, raw_sycl_queues) {
   // Make sure all queues use the same context
+  sycl::queue queue;
+  int* p;
   Kokkos::initialize();
-  Kokkos::SYCL default_space;
-  sycl::context default_context = default_space.sycl_queue().get_context();
-
-  sycl::queue queue(default_context, sycl::default_selector_v,
-                    sycl::property::queue::in_order());
-  int* p            = sycl::malloc_device<int>(100, queue);
-  using MemorySpace = typename TEST_EXECSPACE::memory_space;
-
   {
-    TEST_EXECSPACE space0(queue);
-    Kokkos::View<int*, TEST_EXECSPACE> v(p, 100);
-    Kokkos::deep_copy(space0, v, 5);
-    int sum = 0;
+    Kokkos::SYCL default_space;
+    sycl::context default_context = default_space.sycl_queue().get_context();
 
-    Kokkos::parallel_for("Test::sycl::raw_sycl_queue::Range",
-                         Kokkos::RangePolicy<TEST_EXECSPACE>(space0, 0, 100),
-                         FunctorRange<MemorySpace>(v));
-    Kokkos::parallel_reduce("Test::sycl::raw_sycl_queue::RangeReduce",
-                            Kokkos::RangePolicy<TEST_EXECSPACE>(space0, 0, 100),
-                            FunctorRangeReduce<MemorySpace>(v), sum);
-    space0.fence();
-    ASSERT_EQ(6 * 100, sum);
+    queue             = sycl::queue(default_context, sycl::default_selector_v,
+                                    sycl::property::queue::in_order());
+    p                 = sycl::malloc_device<int>(100, queue);
+    using MemorySpace = typename TEST_EXECSPACE::memory_space;
 
-    Kokkos::parallel_for("Test::sycl::raw_sycl_queue::MDRange",
-                         Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>(
-                             space0, {0, 0}, {10, 10}),
-                         FunctorMDRange<MemorySpace>(v));
-    space0.fence();
-    Kokkos::parallel_reduce(
-        "Test::sycl::raw_sycl_queue::MDRangeReduce",
-        Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>(space0, {0, 0},
-                                                               {10, 10}),
-        FunctorMDRangeReduce<MemorySpace>(v), sum);
-    space0.fence();
-    ASSERT_EQ(7 * 100, sum);
+    {
+      TEST_EXECSPACE space0(queue);
+      Kokkos::View<int*, TEST_EXECSPACE> v(p, 100);
+      Kokkos::deep_copy(space0, v, 5);
+      int sum = 0;
 
-    Kokkos::parallel_for("Test::sycl::raw_sycl_queue::Team",
-                         Kokkos::TeamPolicy<TEST_EXECSPACE>(space0, 10, 10),
-                         FunctorTeam<MemorySpace, TEST_EXECSPACE>(v));
-    space0.fence();
-    Kokkos::parallel_reduce("Test::sycl::raw_sycl_queue::Team",
-                            Kokkos::TeamPolicy<TEST_EXECSPACE>(space0, 10, 10),
-                            FunctorTeamReduce<MemorySpace, TEST_EXECSPACE>(v),
-                            sum);
-    space0.fence();
-    ASSERT_EQ(8 * 100, sum);
+      Kokkos::parallel_for("Test::sycl::raw_sycl_queue::Range",
+                           Kokkos::RangePolicy<TEST_EXECSPACE>(space0, 0, 100),
+                           FunctorRange<MemorySpace>(v));
+      Kokkos::parallel_reduce(
+          "Test::sycl::raw_sycl_queue::RangeReduce",
+          Kokkos::RangePolicy<TEST_EXECSPACE>(space0, 0, 100),
+          FunctorRangeReduce<MemorySpace>(v), sum);
+      space0.fence();
+      ASSERT_EQ(6 * 100, sum);
+
+      Kokkos::parallel_for(
+          "Test::sycl::raw_sycl_queue::MDRange",
+          Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>(space0, {0, 0},
+                                                                 {10, 10}),
+          FunctorMDRange<MemorySpace>(v));
+      space0.fence();
+      Kokkos::parallel_reduce(
+          "Test::sycl::raw_sycl_queue::MDRangeReduce",
+          Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>(space0, {0, 0},
+                                                                 {10, 10}),
+          FunctorMDRangeReduce<MemorySpace>(v), sum);
+      space0.fence();
+      ASSERT_EQ(7 * 100, sum);
+
+      Kokkos::parallel_for("Test::sycl::raw_sycl_queue::Team",
+                           Kokkos::TeamPolicy<TEST_EXECSPACE>(space0, 10, 10),
+                           FunctorTeam<MemorySpace, TEST_EXECSPACE>(v));
+      space0.fence();
+      Kokkos::parallel_reduce(
+          "Test::sycl::raw_sycl_queue::Team",
+          Kokkos::TeamPolicy<TEST_EXECSPACE>(space0, 10, 10),
+          FunctorTeamReduce<MemorySpace, TEST_EXECSPACE>(v), sum);
+      space0.fence();
+      ASSERT_EQ(8 * 100, sum);
+    }
   }
   Kokkos::finalize();
 

